src/main/java/com/google/jenkins/plugins/k8sengine/KubernetesEnginePublisher.java

```diff
@@ -315,15 +338,57 @@ public FormValidation doCheckZone(
       return FormValidation.ok();
     }
 
-    public FormValidation doCheckProjectId(@QueryParameter String value) {
-      if (Strings.isNullOrEmpty(value)) {
+    public ListBoxModel doFillProjectIdItems(
+        @AncestorInPath Jenkins context,
+        @QueryParameter("credentialsId") final String credentialsId) {
+      ListBoxModel items = new ListBoxModel();
+      items.add("");
+      if (Strings.isNullOrEmpty(credentialsId)) {
+        return items;
+      }
+
+      ClientFactory clientFactory;
+      try {
+        clientFactory = this.getClientFactory(context, credentialsId);
+      } catch (AbortException ae) {
+        items.clear();
+        items.add(ae.getMessage(), "");
+        return items;
+      }
+
+      String defaultProjectId = clientFactory.getDefaultProjectId();
+      try {
+        CloudResourceManagerClient client = clientFactory.cloudResourceManagerClient();
+        List<Project> projects = client.getAccountProjects();
+
+        if (!projects.isEmpty()) {
+          projects
+              .stream()
+              .filter(p -> !p.getProjectId().equals(defaultProjectId))
+              .forEach(p -> items.add(p.getProjectId()));
+          if (projects.size() > items.size() - 1 && !Strings.isNullOrEmpty(defaultProjectId)) {
+            items.add(new Option(defaultProjectId, defaultProjectId, true));
+          } else {
+            // Select not the first (empty) item but the second item, which exists because
+            // projects is not empty.
+            items.get(1).selected = true;
+          }
+        }
+        return items;
+      } catch (IOException ioe) {
+        items.add(new Option(defaultProjectId, defaultProjectId, true));
+        return items;
+      }
+    }
+
+    public FormValidation doCheckProjectId(@QueryParameter("projectId") String projectId) {
+      if (Strings.isNullOrEmpty(projectId)) {
```

Comment:
With the new RSM client, this enables us to do a more thorough validation of this field, which will come in handy for the DSL case. Suggest adding logic to validate this field with the RSM client, (if the credId is present).

Reply:
I can add that to this PR or to another PR, similar to the changes for verifying zone.

Reply:
SGTM